azure/scope/machine.go

```diff
@@ -451,22 +451,14 @@ func (m *MachineScope) Role() string {
 	return infrav1.Node
 }
 
-// GetVMID returns the AzureMachine instance id by parsing Spec.FakeProviderID.
+// GetVMID returns the AzureMachine instance id by parsing the scope's providerID.
 func (m *MachineScope) GetVMID() string {
-	parsed, err := noderefutil.NewProviderID(m.ProviderID())
-	if err != nil {
-		return ""
-	}
-	return parsed.ID()
+	return utilazure.GetVMID(m.ProviderID())
```

Comment:
same thing here

Reply:
For this one we need to extract the VM ID so we do need to parse it. Could we use https://github.com/Azure/azure-sdk-for-go/blob/a688cfe08303253b05efb19a7645fbdb24d7375e/sdk/azcore/arm/resource_identifier.go#L20 instead of writing our own parsing function?